### PR TITLE
testing: split system tests

### PR DESCRIPTION
**Pull request by tmatsuo**
I manually run synthtool with the changes in https://github.com/googleapis/synthtool/pull/712

Motivation: #191 

Note that noxfile.py is skipped, so I manually updated it (probably ok)

### CODE REVIEW

**Comment by tmatsuo** [comment]
@tseaver @shollyman @busunkim96 I can not assign a reviewer or add a label. Can you review?

**Comment by tmatsuo** [comment]
I checked the test logs and they work as expected:

* main presubmit build is skipping system tests
* system-x.x builds are only running a system test for a single python version
* presubmit is the slowest. It took 24min 40secs.

Another observation is that the snippets tests are taking long time. Once we move them out, the build will be much faster.

File: noxfile.py [hunk 1]
```diff
@@ -190,3 +194,36 @@ def docs(session):
...
+            "sphinx.ext.todo,"
+            "sphinx.ext.viewcode,"
+            "recommonmark"
+        ),
+        "-b",
+        "html",
+        "-d",
+        os.path.join("docs", "_build", "doctrees", ""),
+        os.path.join("docs", ""),
+        os.path.join("docs", "_build", "html", ""),
```
Comment (tseaver) [inline noxfile.py:228]: Is this supposed to build into the same output directory as the main docs build?

File: synth.py [hunk 2]
```diff
@@ -59,7 +59,7 @@
 # ----------------------------------------------------------------------------
 # Add templated files
 # ----------------------------------------------------------------------------
-templated_files = common.py_library(cov_level=100, samples=True)
+templated_files = common.py_library(cov_level=100, samples=True, split_system_tests=True)
 
 # BigQuery has a custom multiprocessing note
 s.move(templated_files, excludes=["noxfile.py", "docs/multiprocessing.rst"])
```
Comment (tseaver) [inline synth.py:65]: I think we need to exclude `docs/conf.py` as well, to prevent future synth failures.

File: docs/conf.py [hunk 3]
```diff
@@ -20,6 +20,9 @@
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 sys.path.insert(0, os.path.abspath(".."))
 
+# For plugins that can not read conf.py.
+sys.path.insert(0, os.path.abspath("."))
```
Comment (tseaver) [inline docs/conf.py:24]: Is this a workaround for a `sphinx-docfx-yaml` bug?  Should we link that issue here?

File: noxfile.py (hunk 1 above)
Comment (tmatsuo) [reply noxfile.py:228]: For docfx build, we only upload yaml files, so I don't think it matters.

File: synth.py (hunk 2 above)
Comment (tmatsuo) [reply synth.py:65]: I can do that, but I think we should rather fix this by modifying the template to include the excludes. WDYT?

File: docs/conf.py (hunk 3 above)
Comment (tmatsuo) [reply docs/conf.py:24]: I will add it in the template.

**Comment by tmatsuo** [comment]
@tseaver Thanks, I opened https://github.com/googleapis/synthtool/pull/713 which will allow us to use the template.

File: synth.py (hunk 2 above)
Comment (tmatsuo) [reply synth.py:65]: I updated the template in synthtool and re-generated the files. Now we don't manually modify the files.

**Comment by tmatsuo** [comment]
@tseaver Tests are passing. I rebased to the master. PTAL

**Pull request merged**